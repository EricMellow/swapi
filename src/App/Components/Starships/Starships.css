.container {
  width: 80%;
  margin: 30px auto 10px auto;
  display: flex;
  justify-content: space-around;
  height: 65%;
  border: 2px solid white;
  box-shadow: 0px 0px 75px white;
  border-radius: 5px;
  flex-wrap: wrap;
  overflow-y: scroll;
}

.wait {
  font-family: 'Orbitron', sans-serif;
  font-size: 45px;
  text-align: center;
  margin-top: 30px;
  width: 80%;
}

.card {
  width: 220px;
  height: 275px;
  border: 2px solid #882233;
  overflow-y: scroll;
  text-align: center;
  margin: 10px;
  border-radius: 5px;
  background-color: rgb(255, 255, 255, .3);
}

p {
  font-size: 18px;
  margin: 15px 5px;
}

h3 {
  font-size: 22px;
  margin: 5px;
}