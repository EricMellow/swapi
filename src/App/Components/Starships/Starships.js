import React from 'react';
import './Starships.css';

const Starships = ({cardData}) => {
  let starshipCards = [];
  if (cardData) {
    starshipCards = cardData.map((starship, index) => {
      return (
        <div key={starship.name + index} className="card">
          <h3>{starship.name}</h3>
          <p>Model: {starship.model}</p>
          <p>Hyperdrive Rating: {starship.hyperdriveRating}</p>
          <p>Number of Passengers: {starship.numOfPassengers}</p>
        </div>
      )
    })

  }

  return (
    <div className="container">
      {starshipCards.length === 0 && <div className="wait">Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.</div>}
      {starshipCards.length > 1 && starshipCards}
    </div>
  );
};

export default Starships;