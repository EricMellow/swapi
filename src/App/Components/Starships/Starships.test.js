import React from "react";
import Starships from "./Starships";
import { shallow } from "enzyme";
import { mockCleanedStarships } from '../../mockStarshipData';

describe('Starships', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Starships />);
  });

  it('should match the snapshot with no data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with good card data', () => {
    const mockData = mockCleanedStarships;
    wrapper = shallow(<Starships cardData={mockData}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with error data', () => {
    const mockError = [{
      hyperdriveRating: ":(",
      model: "Something went wrong",
      name: "Oh no!",
      numOfPassengers: "No data"
    }];
    wrapper = shallow(<Starships cardData={mockError} />);

    expect(wrapper).toMatchSnapshot();
  });


});