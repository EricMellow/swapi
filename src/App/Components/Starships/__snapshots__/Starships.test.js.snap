// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Starships should match the snapshot with error data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Starships
    cardData={
        Array [
            Object {
              "hyperdriveRating": ":(",
              "model": "Something went wrong",
              "name": "Oh no!",
              "numOfPassengers": "No data",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Starships should match the snapshot with good card data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Starships
    cardData={
        Array [
            Object {
              "hyperdriveRating": "2.0",
              "model": "Executor-class star dreadnought",
              "name": "Executor",
              "numOfPassengers": "38000",
            },
            Object {
              "hyperdriveRating": "2.0",
              "model": "Cool Starship",
              "name": "Millenium Falcon",
              "numOfPassengers": "6",
            },
            Object {
              "hyperdriveRating": "2.0",
              "model": "Fighter",
              "name": "X-Wing",
              "numOfPassengers": "0",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        Array [
          <div
            className="card"
>
            <h3>
                        Executor
            </h3>
            <button
                        className="fav"
                        onClick={[Function]}
            />
            <p>
                        Model: 
                        Executor-class star dreadnought
            </p>
            <p>
                        Hyperdrive Rating: 
                        2.0
            </p>
            <p>
                        Number of Passengers: 
                        38000
            </p>
</div>,
          <div
            className="card"
>
            <h3>
                        Millenium Falcon
            </h3>
            <button
                        className="fav"
                        onClick={[Function]}
            />
            <p>
                        Model: 
                        Cool Starship
            </p>
            <p>
                        Hyperdrive Rating: 
                        2.0
            </p>
            <p>
                        Number of Passengers: 
                        6
            </p>
</div>,
          <div
            className="card"
>
            <h3>
                        X-Wing
            </h3>
            <button
                        className="fav"
                        onClick={[Function]}
            />
            <p>
                        Model: 
                        Fighter
            </p>
            <p>
                        Hyperdrive Rating: 
                        2.0
            </p>
            <p>
                        Number of Passengers: 
                        0
            </p>
</div>,
        ],
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": "Executor0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Executor
</h3>,
            <button
              className="fav"
              onClick={[Function]}
/>,
            <p>
              Model: 
              Executor-class star dreadnought
</p>,
            <p>
              Hyperdrive Rating: 
              2.0
</p>,
            <p>
              Number of Passengers: 
              38000
</p>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Executor",
            },
            "ref": null,
            "rendered": "Executor",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Model: ",
                "Executor-class star dreadnought",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Model: ",
              "Executor-class star dreadnought",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Hyperdrive Rating: ",
                "2.0",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Hyperdrive Rating: ",
              "2.0",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Number of Passengers: ",
                "38000",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Number of Passengers: ",
              "38000",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Millenium Falcon1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Millenium Falcon
</h3>,
            <button
              className="fav"
              onClick={[Function]}
/>,
            <p>
              Model: 
              Cool Starship
</p>,
            <p>
              Hyperdrive Rating: 
              2.0
</p>,
            <p>
              Number of Passengers: 
              6
</p>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Millenium Falcon",
            },
            "ref": null,
            "rendered": "Millenium Falcon",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Model: ",
                "Cool Starship",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Model: ",
              "Cool Starship",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Hyperdrive Rating: ",
                "2.0",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Hyperdrive Rating: ",
              "2.0",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Number of Passengers: ",
                "6",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Number of Passengers: ",
              "6",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "X-Wing2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              X-Wing
</h3>,
            <button
              className="fav"
              onClick={[Function]}
/>,
            <p>
              Model: 
              Fighter
</p>,
            <p>
              Hyperdrive Rating: 
              2.0
</p>,
            <p>
              Number of Passengers: 
              0
</p>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "X-Wing",
            },
            "ref": null,
            "rendered": "X-Wing",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fav",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Model: ",
                "Fighter",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Model: ",
              "Fighter",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Hyperdrive Rating: ",
                "2.0",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Hyperdrive Rating: ",
              "2.0",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Number of Passengers: ",
                "0",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Number of Passengers: ",
              "0",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          Array [
            <div
              className="card"
>
              <h3>
                            Executor
              </h3>
              <button
                            className="fav"
                            onClick={[Function]}
              />
              <p>
                            Model: 
                            Executor-class star dreadnought
              </p>
              <p>
                            Hyperdrive Rating: 
                            2.0
              </p>
              <p>
                            Number of Passengers: 
                            38000
              </p>
</div>,
            <div
              className="card"
>
              <h3>
                            Millenium Falcon
              </h3>
              <button
                            className="fav"
                            onClick={[Function]}
              />
              <p>
                            Model: 
                            Cool Starship
              </p>
              <p>
                            Hyperdrive Rating: 
                            2.0
              </p>
              <p>
                            Number of Passengers: 
                            6
              </p>
</div>,
            <div
              className="card"
>
              <h3>
                            X-Wing
              </h3>
              <button
                            className="fav"
                            onClick={[Function]}
              />
              <p>
                            Model: 
                            Fighter
              </p>
              <p>
                            Hyperdrive Rating: 
                            2.0
              </p>
              <p>
                            Number of Passengers: 
                            0
              </p>
</div>,
          ],
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": "Executor0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Executor
</h3>,
              <button
                className="fav"
                onClick={[Function]}
/>,
              <p>
                Model: 
                Executor-class star dreadnought
</p>,
              <p>
                Hyperdrive Rating: 
                2.0
</p>,
              <p>
                Number of Passengers: 
                38000
</p>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Executor",
              },
              "ref": null,
              "rendered": "Executor",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Model: ",
                  "Executor-class star dreadnought",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Model: ",
                "Executor-class star dreadnought",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Hyperdrive Rating: ",
                  "2.0",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Hyperdrive Rating: ",
                "2.0",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Number of Passengers: ",
                  "38000",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Number of Passengers: ",
                "38000",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Millenium Falcon1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Millenium Falcon
</h3>,
              <button
                className="fav"
                onClick={[Function]}
/>,
              <p>
                Model: 
                Cool Starship
</p>,
              <p>
                Hyperdrive Rating: 
                2.0
</p>,
              <p>
                Number of Passengers: 
                6
</p>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Millenium Falcon",
              },
              "ref": null,
              "rendered": "Millenium Falcon",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Model: ",
                  "Cool Starship",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Model: ",
                "Cool Starship",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Hyperdrive Rating: ",
                  "2.0",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Hyperdrive Rating: ",
                "2.0",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Number of Passengers: ",
                  "6",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Number of Passengers: ",
                "6",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "X-Wing2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                X-Wing
</h3>,
              <button
                className="fav"
                onClick={[Function]}
/>,
              <p>
                Model: 
                Fighter
</p>,
              <p>
                Hyperdrive Rating: 
                2.0
</p>,
              <p>
                Number of Passengers: 
                0
</p>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "X-Wing",
              },
              "ref": null,
              "rendered": "X-Wing",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fav",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Model: ",
                  "Fighter",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Model: ",
                "Fighter",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Hyperdrive Rating: ",
                  "2.0",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Hyperdrive Rating: ",
                "2.0",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Number of Passengers: ",
                  "0",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Number of Passengers: ",
                "0",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Starships should match the snapshot with no data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Starships />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="wait"
>
          Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.
</div>,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
          "className": "wait",
        },
        "ref": null,
        "rendered": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="wait"
>
            Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.
</div>,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
            "className": "wait",
          },
          "ref": null,
          "rendered": "Access code YT-1300. Retreiving your SWAPIBOX experience. Please Wait.",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
